### DISCUSSION TITLE
Custom body slot for grid mode Table Component

### DISCUSSION THREAD

**Discussion by waynemamahit** [quasarframework/quasar#12561] [Ideas / Proposals]
Could i get custom body slot for grid mode on table component? I think this is usefull feature on table component

**Comment by MilosPaunovic** [comment]
You can use regular `body` slot and differentiate whatever you need using screen plugin:
[https://quasar.dev/options/screen-plugin](https://quasar.dev/options/screen-plugin)

**Comment by metalsadman** [comment]
Yes via making a wrapper of QTable, using `item` slot, then implement your custom grid slot inside it.

**Comment by meo1** [comment]
I am sorry, I could not understand your solutions. I have very similar problem. Can you explain it in detail, I send you also a fragment of my code, (I am sorry, I could not send as code fragment, there is a bug in Github's editor): 

![image](https://user-images.githubusercontent.com/35503088/180600655-2fc73307-3c34-45fb-821d-1f040b3a75e9.png)

**Reply by meo1** [reply]
`<q-table :rows="DataStore.ObjectRows" :columns="DataStore.ObjectsVisibleColumns"  row-key="id" :grid="store.userDB.view=='tile'" dense>

        <template v-slot:body="props">
          <q-tr :props="props">
            <q-td v-for="item in DataStore.ObjectsVisibleColumns">
              <div :props="props" v-if="item.name=='image_small'">
                <q-img :src="DataStore.FileURL + props.row[item.name] + '?w=40'" style="max-width: 40px">
                  <q-tooltip>
                    <img :src="DataStore.FileURL + props.row.image_big + '?w=256'" />
                  </q-tooltip>
                </q-img>
              </div>
              <div :props="props" v-else-if="item.name=='private'">
                <q-checkbox v-model="props.row.private" disable />
              </div>
              <div :props="props" v-else>{{props.row[item.name]}} </div>
            </q-td>
          </q-tr>
        </template>
     
 </q-table>`

**Reply by meo1** [reply]
`<q-table :rows="DataStore.ObjectRows" :columns="DataStore.ObjectsVisibleColumns"  row-key="id" :grid="store.userDB.view=='tile'" dense>`

**Reply by meo1** [reply]
I could only manage send the code partially because of the bug of the github.

**Reply by meo1** [reply]
The following is how my table looks like in list view:

![image](https://user-images.githubusercontent.com/35503088/180600962-1bbd5be8-316d-469f-b658-3662e73f752b.png)

And the following picture is in grid view:

![image](https://user-images.githubusercontent.com/35503088/180601007-6c7398c9-8004-42f0-820b-f64a92144fad.png)

**Comment by meo1** [comment]
OK. I finally understood your ( @metalsadman ) solution. That will solve my problem.

**Reply by metalsadman** [reply]
See this example also https://codepen.io/metalsadman/pen/ZgKexK

**Comment by nekooee** [comment]
I noticed today that my buttons are not visible in grid mode. Not only the buttons but also the options that are true and false disappear. I am utterly disappointed with Quasar. What is the reason I write so much additional code for the grid mode in every part of the data table? 
At first, I thought I would find a simple solution for it with a search. But in disbelief, I realized the complexity of this issue.
😒😒

In the end, I gave up the grade for now. Hopefully, in the next updates, there will be no need for so many additional codes for a simple task.
